### DISCUSSION TITLE
How to rewrite for specific client ?

### DISCUSSION THREAD

**Discussion by rayures** [AdguardTeam/AdGuardHome#3630] [Q&A] [answered]
hi, could not find a working answer in the issues or wiki pages. 

how do i setup custom dns rewrites for a specific clients list (settings / client settings )?

--
i have setup 1 adguard home 0.106.3server for internal and external use. external via dot.

external clients are matched via IP (everything not 192.168.*.*) this works fine.
i have setup dns rewrites for internal clients to internal ip addresses, and i want the external clients to not use them. 

i tried different things, ie. a custom filtering rule with the $dnsrewrite modifier, but it looks like dns rewrites always win. How to set this up?

tia.

//edit. 

tried some things. the custom rewrite rule works, when the dns rewrite entry is removed. So 'dns rewrites' overwrite custom rules.

//edit2: 
-removed dns rewrite rule and added 2 custom rules. that works.

||mail.example.org^$important,client='external',dnsrewrite=example.org
||mail.example.org^$dnsrewrite=192.168.x.x

**Comment by ameshkov** [comment, marked as answer]
I see you've been able to figure this out already, `client` and `dnsrewrite` is indeed what you need.

You can also use the `ctag` modifier if you want to limit the rule to a group of clients:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#ctag

**Comment by jwgn** [comment]
How to modify the DNS reply code for specific clients? For example, answer with NXDOMAIN in general, but with REJECT for a specific client?

**Reply by jwgn** [reply]
Found the answer here https://adguard-dns.io/kb/general/dns-filtering-syntax/
